src/foam/lib/json/FObjectParser.java

```diff
@@ -46,7 +46,13 @@ public PStream parse(PStream ps, ParserContext x) {
               subx.set("enum", c);
               subParser = EnumParserFactory.getInstance(c);
             } else {
-              Object obj = ((X) x.get("X")).create(c);
+              Object obj = null;
```

Comment:
Why do we do this rather than just always provide the X when using FObjectParser?

Reply:
Can you explain more?

Reply:
Please follow-up with Kevin directly to address this comment.